File: HackerRank/RepeatedString.cpp
#include <bits/stdc++.h>

using namespace std;

// Complete the repeatedString function below.
long repeatedString(string s, long n)
{
    long cnt = 0;
    long aCnt = 0;
    string newStr = "";
    for (long i = 0; i < s.size(); i++)
    {
        if (s[i] == 'a')
            aCnt++;
    }

    long repeat1 = n / s.size();
    long repeat2 = n % s.size();

    cnt = aCnt * repeat1;
    for (int i = 0; i < repeat2; i++)
    {
        if (s[i] == 'a')
            cnt++;
    }

    return cnt;
}

int main()
{
    ofstream fout(getenv("OUTPUT_PATH"));

    string s;
    getline(cin, s);

    long n;
    cin >> n;
    cin.ignore(numeric_limits<streamsize>::max(), '\n');

    long result = repeatedString(s, n);

    fout << result << "\n";

    fout.close();

    return 0;
}

File: HackerRank/RepeatedString.py
# Repeated String
#!/bin/python3

import math
import os
import random
import re
import sys

# Complete the repeatedString function below.

'''
s = input().strip()
n = int(input().strip())
L = len(s)
print(s.count('a') * (n//L) + s[:n % L].count('a'))
'''


def repeatedString(s, n):
    aCnt = s.count('a')
    cnt = aCnt * (n // len(s))
    '''
    for i in range(n % len(s)):
        if s[i] == 'a':
            cnt += 1
    '''
    cnt += s[:n % len(s)].count('a')
    return cnt


if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    s = input()

    n = int(input())

    result = repeatedString(s, n)

    fptr.write(str(result) + '\n')

    fptr.close()
